### DISCUSSION TITLE
Build target x86

### DISCUSSION THREAD

**Discussion by Sven-Moennich** [Freetz-NG/freetz-ng#533] [Allgemein]
Moin.
Ich hätte eine frage zum build.
Und zwar ist es so das ich alles was momentan cross compiliert wird für die entsprechende hardware für x86 gebaut haben möchte.
Das was im image nicht durch freetz stuff ersetzt wird dürfte erst mal bleiben.

Gibts da nen tip wo ich da eingreifen kann ?

**Comment by fda77** [comment]
Ich versteh die Frage nicht so ganz. Jedenfalls wenn du für x86 bauen willst musst du ein Gerät auswählen das dieses hat. Das sind die Docsis/Cable boxen

**Reply by Sven-Moennich** [reply]
Ok, vllt ein wenig missverständlich ausgedrückt.
Ich möchte zb ein image einer 7590 nehmen und alles was geht für einen raspi kompilieren.
Das image was gebaut werden soll soll nicht für eine fritzbox sein.
Mir ist klar das sich aus nem pc oder raspi keine fritzbox machen lässt.

**Reply by pfichtner** [reply]
Mir ist völlig unklar was Du vor hast und befürchte Du hast etwas geplant, was nicht funktionieren kann/wird. 
Kannst Du mal erklären was Du machen möchtest bzw. was am Ende raus kommen soll, was Du erreichen möchtest?

**Reply by Sven-Moennich** [reply]
Ich weis auch nicht wie ichs erklären soll.
Eigentlich will ich nur ein Image der 7590 auf nem Raspi zum laufen bekommen.
Ich hatte da schon mal angefangen mein eigenes tool zu basteln wodurch ich dann auf freetz aufmerksam wurde und mir gedacht hab das es vllt einfacher wäre als mich allein durch den ganzen AVM quatsch zu try n errorn.
Aber egal.
Freetz kann nur Images für AVM Devices bauen. Das ist nicht das was ich vor hab.
Erstes ziel sollte sein das erst mal überhaupt einigermaßen zum laufen zu bekommen.
Alles was proprietär von avm kommt würde ich nach bauen.
Zweites ziel sollte dann sein das fritz os modulartig anpassbar zu machen. ähnlich wie freetz, mehr so wie avms baukastensystem.
Daher war meine Frage wie ich das build target ändern könnte. Freetz baut sich das aus ziemlich vielen abhängigkeiten zusammen so wie ich das sehe. Eigentlich wollt ich durch die Frage die ganze bastelei ein wenig abkürzen, aber mittlerweile schein ich mehr erklären zu müssen als ich gedacht hab.
BTW bei Sätzen wie nicht funktionieren kann/wird mit bezug auf digitaltechnik bringen mich nahe an nen mehrstündigen lachanfall.

**Reply by pfichtner** [reply]
> BTW bei Sätzen wie nicht funktionieren kann/wird mit bezug auf digitaltechnik bringen mich nahe an nen mehrstündigen lachanfall.

Wenn der Lachanfall nachlässt, kannst Du gerne erklären, wie Du z.B. DECT auf dem Raspi ans Laufen kriegen willst. Bis gerade eben war mir nicht klar, dass dies an meinen mangelnden Digitaltechnik-Kenntnissen scheitert. 
Ich korrigiere: Auch das ist nicht unmöglich, das wollen wir aber sicher nicht ernsthaft diskutieren, ob bzw. wie das geht und wleche Aufwände dies erzeugen würde und was eine mögliche Lösung dann tatsächlich technisch leisten würde und ob das nicht eher ein proof of concept ist.

**Reply by pfichtner** [reply]
Freetz kompiliert ja "nur" die eigenen Dinge, das Image wird ja dann aus dem original Image und dem gebauten gemergt. Es bringt Dir ja nichts, wenn Freetz für x86 kompiliert ist der Rest und damit wichtige und notwendige Teile des Systems aber nicht für x86 sind. Oder habe ich da was falsch verstanden? Willst Du wirklich das Image booten? Oder Dein Grundsystem bei raspian o.ä. belassen und dann Teile aus dem Image nativ ausführen?

**Reply by Sven-Moennich** [reply]
> das wollen wir aber sicher nicht ernsthaft diskutieren
Das war mit nen Grund warum ich die Diskussion nicht "Ist es sinnvoll einen Raspi als DECT Basis zu betreiben?" genannt hab.

> Freetz kompiliert ja "nur" die eigenen Dinge, das Image wird ja dann aus dem original Image und dem gebauten gemergt.
Genau.

> Es bringt Dir ja nichts, wenn Freetz für x86 kompiliert ist der Rest und damit wichtige und notwendige Teile des Systems aber nicht für x86 sind. Oder habe ich da was falsch verstanden?

Nö richtig. Das war es was ich mit:
> Das was im image nicht durch freetz stuff ersetzt wird dürfte erst mal bleiben.
und

> Alles was proprietär von avm kommt würde ich nach bauen.
zum ausdruck bringen wollte.

> Willst Du wirklich das Image booten?
So gesehen ja. Wobei das jetzt mit nem Raspi weiter hinten auf der Liste steht. Weil da doch einiges mehr vom original abgewichen werden müsste. 

> Oder Dein Grundsystem bei raspian o.ä. belassen und dann Teile aus dem Image nativ ausführen?
Ne, eher wirklich image booten. 

Ganz eigentlich wollt ich eigentlich nur das frontend portierfähig machen. aber das erschien mir nicht aufregend genug.
Da ich mit meinen fb_tools zum auspacken packen usw von images mittlerweile soweit bin hab ich mir die frage gestellt ob ich mit meinen tools weiter mache oder ob ich das in freetz einarbeite. beides würde wahrscheinlich gleich lang dauern. Davon ausgehend das man bei freetz ggf ein wenig hilfe bekommt, dürfte das ein wenig weniger Lebenszeit kosten. Wobei ich mir da jetzt keine großen hoffnungen gemacht habe. Bei meinen googleleien zum thema wurde mir ja auch hin und wieder das ipp forum angeboten. Was ich dann auch kurzerhand aus meiner ergebnisliste gelöscht hab. und ich hatte eigentlich damit gerechnet das von hier dahin verwiesen wird.
Da sich das hier allerdings nicht groß vom forum zu unterscheiden scheint hab ich mich dazu entschlossen das die welt mit nen paar mehr tools schon klar kommen wird und ich mein eigenes Projekt weiter verfolgen werde. Von daher, nix für ungut, viel spass noch.

**Comment by fda77** [comment]
Kein Raspberry hat x86.
Binaries mit Freetz gebaut sind für Raspberry eh nicht sonderlich geeignet da Packages anpassungen haben können, zb der Pfad zum speichern.
Wie wäre es wenn du gcc etc auf dem Rpi installierst und direkt dort compilierst?

**Reply by Sven-Moennich** [reply]
Irgendwas muss mächtig schief gelaufen sein, wenn man mir nen Raspi erklären will.
Bisher hab ich die Sachen manuell compiliert und ins dateisystem gequetscht. Das ist allerdings nicht das was ich von einem Computer erwarte. Der sollte das nach möglichkeit automatisch können. Ich glaube das war auch bei freetz son hintergedanke der dazu geführt hat das man nicht jedes paket einzeln compilieren muss und händisch ins image friemeln muss. Aber ich kann mich auch irren.
Mir ist schon klar das freetz eigentlich für was anderes gedacht ist.

**Reply by fda77** [reply]
Wen du alles weist, warum frags du?
